File: Kotlin/프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.kt
import kotlin.math.*

class Solution {
    fun solution(n: Int): Int {
        var answer : Int = 0
        var nn = n

        val sb = StringBuilder()

        while (nn > 0) {
            sb.append(nn % 3)
            println(sb.toString())
            nn /= 3
        }

        val charArr = sb.reverse().toString().toCharArray()

        for (i in charArr.indices) {
            val c = charArr[i].digitToInt()

            answer += c * (3.0.pow(i)).toInt()
        }

        return answer
    }
}

File: C#/프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cs
using System;

public class Solution {
    public int solution(int n) {
        int answer = 0;
        //15... 0
        //5... 0
        //1... 2
        //0... 1
        
        string s = "";
        while (n != 0)
        {
            int t = n % 3;
            n /= 3;
            
            s += t.ToString();
        }
        
        for (int i = 0; i < s.Length; i++)
        {
            answer += (int)Math.Pow(3, s.Length - i - 1) * (s[i] - '0');
        }
        
        return answer;
    }
}